reaper/reaper.go

```diff
@@ -42,13 +42,13 @@ func NewReaper(cf cloudfoundry.Client, currentTime func() time.Time, output io.W
 	}
 }
 
-func (r Reaper) Reap(serviceName string, expiryInterval time.Duration, reap, recursive bool) error {
+func (r Reaper) Reap(serviceName string, planName string, expiryInterval time.Duration, reap, recursive bool) error {
 	r.expiryInterval = expiryInterval
 	r.reap = reap
 	r.recursive = recursive
 	r.errorChannel = make(chan error, 100)
 
-	r.delete(r.expiredInstancesOf(r.freeServicePlansOf(r.servicesWithName(serviceName))))
+	r.delete(r.expiredInstancesOf(r.ServicePlansWithName(r.servicesWithName(serviceName), planName)))
```

Comment:
Heh, this was pretty weird to look at before, and now it needs another parameter. I wonder if it would make it any clearer to stash the `planName` and `serviceName` in the `Reaper` struct, and then you could do something like:

`r.delete(r.expiredInstancesOf(r.eligibleServicePlansFrom(r.eligibleServices)))`

and then it reads a bit more naturally again. What do you think?

Reply:
I agree that reads weird and might be better to rename it.
However, I wouldn't add it to the struct.

Imagine the usage if we would like to delete several plans. We would have to initialize it once per plan rather than calling the method the times we need.

What do you think?